src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ProductController.php

```diff
@@ -63,6 +70,98 @@ class ProductController extends FrameworkBundleAdminController
      */
     private const PRODUCT_CONTROLLER_PERMISSION = 'ADMINPRODUCTS_';
 
+    /**
+     * Shows products listing.
+     *
+     * @AdminSecurity("is_granted(['read'], request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     * @param ProductFilters $filters
+     *
+     * @return Response
+     */
+    public function indexAction(Request $request, ProductFilters $filters): Response
+    {
+        $productGridFactory = $this->get('prestashop.core.grid.factory.product');
+        $productGrid = $productGridFactory->getGrid($filters);
+
+        return $this->render('@PrestaShop/Admin/Sell/Catalog/Product/index.html.twig', [
+            'productGrid' => $this->presentGrid($productGrid),
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink('AdminProducts'),
+        ]);
+    }
+
+    /**
+     * Provides filters functionality
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function searchAction(Request $request): RedirectResponse
+    {
+        /** @var ResponseBuilder $responseBuilder */
+        $responseBuilder = $this->get('prestashop.bundle.grid.response_builder');
+
+        return $responseBuilder->buildSearchResponse(
+            $this->get('prestashop.core.grid.definition.factory.product'),
+            $request,
+            ProductGridDefinitionFactory::GRID_ID,
+            'admin_products_v2_index'
+        );
+    }
+
+    /**
+     * Redirects to step where product price can be edited.
+     *
+     * @AdminSecurity(
+     *     "is_granted(['update'], request.get('_legacy_controller'))",
+     *     redirectRoute="admin_products_index",
+     *     message="You do not have permission to edit this."
+     * )
+     *
+     * @param int $productId
+     *
+     * @return RedirectResponse
+     */
+    public function editPriceAction(int $productId): RedirectResponse
```

Comment:
Good point about routes. Turns out Symfony router has a thing called fragment, but I needed to make some changes to PrestaShops router because it was removing fragment when adding the token. 

So I have improved LinkColumn to have parameter fragment.